### DISCUSSION TITLE
Archived build failed (Catalyst only) with other library used swift-collections

### DISCUSSION THREAD

**Discussion by marcuswu0814** [pointfreeco/swift-composable-architecture#3525] [General]
### Description

When archiving the test app, the process works fine for the iOS app but results in a compiler error when archiving for Catalyst.

Issue also opened at [MetaCodable](https://github.com/SwiftyLab/MetaCodable/issues/116), not sure is this cause by macro package.

### Checklist

- [X] I have determined whether this bug is also reproducible in a vanilla SwiftUI project.
- [X] If possible, I've reproduced the issue using the `main` branch of this package.
- [X] This issue hasn't been addressed in an [existing GitHub issue](https://github.com/pointfreeco/swift-composable-architecture/issues) or [discussion](https://github.com/pointfreeco/swift-composable-architecture/discussions).

### Expected behavior

Like archived iOS app, build/archived success

### Actual behavior

Build failed, showing error like:

https://github.com/marcuswu0814/MetaCodableArchivedIssue/blob/main/CleanShot%202024-11-28%20at%2011.23.33@2x.png

### Reproducing project

https://github.com/marcuswu0814/MetaCodableArchivedIssue

### The Composable Architecture version information

1.16.1

### Destination operating system

iOS 18

### Xcode version information

Xcode 16.2 beta 2 (Also not work with Xcode 16.1)

### Swift Compiler version information

```shell
swift-driver version: 1.115.1 Apple Swift version 6.0.3 (swiftlang-6.0.3.1.4 clang-1600.0.30)
```

**Comment by mbrandonw** [comment]
Hi @marcuswu0814, it's not clear to me that this is a TCA-specific problem and not just a problem with Xcode+Catalyst or perhaps even swift-collections. Can you please do some more investigation to provide some evidence that it is definitely something we can even fix in our library?

For the time being I am going to convert this to a discussion. Feel free to continue the conversation over there.
